src/test/regress/regress_gp.c

```diff
@@ -642,7 +646,12 @@ resGroupPalloc(PG_FUNCTION_ARGS)
 		PG_RETURN_INT32(0);
 
 	ResGroupGetMemInfo(&memLimit, &slotQuota, &sharedQuota);
-	size = ceilf(memLimit * ratio);
+	if (!startupConsidered) {
+		size = ceilf(memLimit * ratio) - 12;
+		startupConsidered = true;
+	} else
+		size = ceilf(memLimit * ratio);
+
```

Comment:
Maybe use allman brace style and move simpler case first?
```suggestion
	if (startupConsidered)
		size = ceilf(memLimit * ratio);
	else 
	{
		size = ceilf(memLimit * ratio) - 12;
		startupConsidered = true;
	}
```

Reply:
Yes, the style doesn't match the codebase. But I would leave the order to make it more visible that the first allocation acts differently

Reply:
I fixed the formatting